sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -661,6 +689,37 @@ export class CryptographyClient {
       return options;
     }
   }
+
+  /**
+   * Checks whether the internal key can be used to execute a given operation, by the operation's name.
+   * @param operation The name of the operation that is expected to be viable
+   */
+  private async checkPermissions(operation: KeyOperation): Promise<void> {
+    await this.getLocalCryptographyClient();
+
+    if (typeof this.key !== "string" && this.key.keyOps && !this.key.keyOps.includes(operation)) {
+      throw new Error(`Operation ${operation} is not supported on key ${this.getKeyID()}`);
+    }
+  }
+
+  /**
+   * Checks whether a key can be used at that specific moment,
+   * by comparing the current date with the bundle's notBefore and expires values.
+   */
+  private async checkKeyValidity(): Promise<void> {
+    await this.getLocalCryptographyClient();
+    const attributes = this.keyBundle?.attributes || {};
+    const { notBefore, expires } = attributes;
+    const now = new Date();
+
+    if (notBefore && now < notBefore) {
+      throw new Error(`Key ${this.getKeyID()} can't be used before ${notBefore}`);
+    }
+
+    if (expires && now < expires) {
```

Comment:
Shouldn't this check for `now > expires`? Please be sure to add test coverage for this and `notBefore`, as well as missing `keyOps` for requested operations (e.g. they aren't allowed to sign but call `sign()` anyway).